script/mirror_db.sh

```diff
@@ -4,45 +4,48 @@
 # Useful for when you want to test a migration against production data, or see
 # the effect of codebase changes on real-life data
 
-# Usage: script/mirror_db.sh [ofn-staging1|ofn-staging2|ofn-prod]
-
 set -e
 
+HOST="$1"
+BUCKET="$2"
+: ${DB_USER='ofn_user'}
+: ${DB_DATABASE='openfoodnetwork'}
+
+DB_OPTIONS=(
+--exclude-table-data=sessions
+--exclude-table-data=spree_log_entries
+--exclude-table-data=spree_state_changes
+--no-acl
+)
+
+if [ -z "$HOST" ]; then
+  echo "[Error] SSH host missing.
+Usage: $0 openfoodnetwork@openfoodnetwork.org.au" >&2
+  exit 1
+fi
+
 if hash zeus 2>/dev/null && [ -e .zeus.sock ]; then
   RAILS_RUN='zeus r'
 else
   RAILS_RUN='bundle exec rails runner'
 fi
 
-if [[ $1 != 'ofn-no' ]]; then
-  DB_USER='openfoodweb'
-  DB_DATABASE='openfoodweb_production'
-else
-  DB_USER='ofn_user'
-  DB_DATABASE='openfoodnetwork'
-fi
-
-DB_OPTIONS='--exclude-table-data=sessions'
-
 # -- Mirror database
 echo "Mirroring database..."
-echo "drop database open_food_network_dev" | psql -h localhost -U ofn open_food_network_test
-echo "create database open_food_network_dev" | psql -h localhost -U ofn open_food_network_test
-ssh $1 "pg_dump -h localhost -U $DB_USER $DB_DATABASE $DB_OPTIONS |gzip" |gunzip |psql -h localhost -U ofn open_food_network_dev
-
+dropdb -h localhost -U ofn open_food_network_dev
+createdb -h localhost -U ofn open_food_network_dev
+ssh -C "$HOST" "pg_dump -h localhost -U $DB_USER $DB_DATABASE ${DB_OPTIONS[@]}" | psql -h localhost -U ofn open_food_network_dev
```

Comment:
So you're pulling a production database and adding it to your local machine. Wont the details be mismatched? eg. the owner will be `ofn_user` and the database will be `openfoodnetwork` in the production dump, but that won't match your dev settings, right?

Reply:
Yes, the owner doesn't match but as long as you have the superuser role accessing the database it doesn't matter. The database name doesn't matter either, it's copied from `$DB_DATABASE` to `open_food_network_dev`.